src/Microsoft.OpenApi.OData.Reader/Operation/OperationHandler.cs

```diff
@@ -139,12 +139,15 @@ protected virtual void SetRequestBody(OpenApiOperation operation)
         /// <param name="operation">The <see cref="OpenApiOperation"/>.</param>
         protected virtual void SetParameters(OpenApiOperation operation)
         {
-            foreach (ODataKeySegment keySegment in Path.OfType<ODataKeySegment>())
+            if (!Context.Settings.DeclarePathParametersOnPathItem)
```

Comment:
Why not check for this flag in line 52? `SetParameters(operation);`

Reply:
The reason for doing it here is because `SetParameters(operation)` in `OperationHandler` has additional functionality to create more query and header parameters depending on the operation e.g. $top, $filter etc https://github.com/microsoft/OpenAPI.NET.OData/blob/master/src/Microsoft.OpenApi.OData.Reader/Operation/OperationHandler.cs#L151-L160. This is unlike in `PathItemHandler` where we are only interested in path parameters. So the flag should only be applied to control whether we add path parameters in OperationHandler or not, and not affect the query and header parameters.